### PR TITLE
[Qt] Document QMapbox + qdoc fixes

### LINKED ISSUE #7980: Document qmapbox.hpp
Missing documentation for `platform/qt/include/qmapbox.hpp`.

/cc @tmpsantos

### PR DESCRIPTION
**Pull request by brunoabinader**
Fixes #7980.

### CODE REVIEW

File: platform/qt/src/qmapbox.cpp [hunk 1]
```diff
@@ -24,16 +24,215 @@ static_assert(mbgl::underlying_type(QMapbox::Feature::PolygonType) == mbgl::unde
...
+    \namespace QMapbox
+    \inmodule Mapbox Qt SDK
+
+    Contains miscellaneous Mapbox bindings used throughout QMapboxGL.
+*/
+
+/*!
+    \typedef QMapbox::Coordinate
+
+    Synonim for QPair<double, double>.
```
Comment (tmpsantos) [inline platform/qt/src/qmapbox.cpp]: Synonym

File: platform/qt/src/qmapbox.cpp [hunk 2]
```diff
@@ -24,16 +24,215 @@ static_assert(mbgl::underlying_type(QMapbox::Feature::PolygonType) == mbgl::unde
...
+    Reflects mbgl::Point<double>.
+*/
+
+/*!
+    \typedef QMapbox::CoordinateZoom
+
+    Synonim for QPair<Coordinate, double>.
```
Comment (tmpsantos) [inline platform/qt/src/qmapbox.cpp]: ditto

File: platform/qt/src/qmapbox.cpp [hunk 3]
```diff
@@ -24,16 +24,215 @@ static_assert(mbgl::underlying_type(QMapbox::Feature::PolygonType) == mbgl::unde
...
+    Used as return value in QMapboxGL::coordinateZoomForBounds.
+*/
+
+/*!
+    \typedef QMapbox::Coordinates
+
+    Synonim for QList<QMapbox::Coordinate>.
```
Comment (tmpsantos) [inline platform/qt/src/qmapbox.cpp]: ditto

File: platform/qt/src/qmapbox.cpp [hunk 4]
```diff
@@ -24,16 +24,215 @@ static_assert(mbgl::underlying_type(QMapbox::Feature::PolygonType) == mbgl::unde
...
+    Reflects mbgl::{LineString,LinearRing,MultiPoint}<double>.
+*/
+
+/*!
+    \typedef QMapbox::CoordinatesCollection
+
+    Synonim for QList<QMapbox::Coordinates>.
```
Comment (tmpsantos) [inline platform/qt/src/qmapbox.cpp]: ditto

File: platform/qt/src/qmapbox.cpp [hunk 5]
```diff
@@ -24,16 +24,215 @@ static_assert(mbgl::underlying_type(QMapbox::Feature::PolygonType) == mbgl::unde
...
+
+    Synonim for QPair<Coordinate, double>.
+    Used as return value in QMapboxGL::coordinateZoomForBounds.
+*/
+
+/*!
+    \typedef QMapbox::Coordinates
+
+    Synonim for QList<QMapbox::Coordinate>.
+    Reflects mbgl::{LineString,LinearRing,MultiPoint}<double>.
```
Comment (tmpsantos) [inline platform/qt/src/qmapbox.cpp]: Not sure if we should mention this. The point of the binding is to hide the core implementation.

File: platform/qt/src/qmapbox.cpp [hunk 6]
```diff
@@ -24,16 +24,215 @@ static_assert(mbgl::underlying_type(QMapbox::Feature::PolygonType) == mbgl::unde
...
+    \typedef QMapbox::CoordinatesCollection
+
+    Synonim for QList<QMapbox::Coordinates>.
+    Reflects mbgl::{MultiLineString,Polygon}<double>.
+*/
+
+/*!
+    \typedef QMapbox::CoordinatesCollections
+
+    Synonim for QList<QMApbox::CoordinatesCollection>.
```
Comment (tmpsantos) [inline platform/qt/src/qmapbox.cpp]: ditto

File: platform/qt/src/qmapbox.cpp [hunk 7]
```diff
@@ -24,16 +24,215 @@ static_assert(mbgl::underlying_type(QMapbox::Feature::PolygonType) == mbgl::unde
...
+    Reflects mbgl::MultiPolygon<double>.
+*/
+
+/*!
+    \class QMapbox::Feature
+    \brief Reflects mbgl::Feature.
```
Comment (tmpsantos) [inline platform/qt/src/qmapbox.cpp]: Not sure about this. Imagine how would you interpret this if you have no idea about mbgl core.

File: platform/qt/src/qmapbox.cpp [hunk 8]
```diff
@@ -24,16 +24,215 @@ static_assert(mbgl::underlying_type(QMapbox::Feature::PolygonType) == mbgl::unde
...
+    QMapbox::FillAnnotation is a struct representing a fill annotation, reflecting mbgl::FillAnnotation contents.
+*/
+
+/*!
+    \class QMapbox::StyleSourcedAnnotation
+    \brief Reflects mbgl::StyleSourcedAnnotation.
+
+    \inmodule Mapbox Qt SDK
+
+    QMapbox::StyleSourcedAnnotation is a struct representing a style sourced annotation, reflecting mbgl::StyleSourcedAnnotation contents.
```
Comment (tmpsantos) [inline platform/qt/src/qmapbox.cpp]: What is a "style sourced annotation"? Explain what can be done with it.

File: platform/qt/src/qmapbox.cpp [hunk 9]
```diff
@@ -24,16 +24,215 @@ static_assert(mbgl::underlying_type(QMapbox::Feature::PolygonType) == mbgl::unde
...
+    QMapbox::LineAnnotation is a struct representing a line annotation, reflecting mbgl::LineAnnotation contents.
+*/
+
+/*!
+    \class QMapbox::FillAnnotation
+    \brief Reflects mbgl::FillAnnotation.
+
+    \inmodule Mapbox Qt SDK
+
+    QMapbox::FillAnnotation is a struct representing a fill annotation, reflecting mbgl::FillAnnotation contents.
```
Comment (tmpsantos) [inline platform/qt/src/qmapbox.cpp]: Explain what it is instead what it reflects.

File: platform/qt/src/qmapbox.cpp [hunk 10]
```diff
@@ -24,16 +24,215 @@ static_assert(mbgl::underlying_type(QMapbox::Feature::PolygonType) == mbgl::unde
...
+    QMapbox::SymbolAnnotation is a struct representing a symbol annotation, reflecting mbgl::SymbolAnnotation contents.
+*/
+
+/*!
+    \class QMapbox::LineAnnotation
+    \brief Reflects mbgl::LineAnnotation.
+
+    \inmodule Mapbox Qt SDK
+
+    QMapbox::LineAnnotation is a struct representing a line annotation, reflecting mbgl::LineAnnotation contents.
```
Comment (tmpsantos) [inline platform/qt/src/qmapbox.cpp]: Explain what it is instead what it reflects.

File: platform/qt/src/qmapbox.cpp [hunk 11]
```diff
@@ -24,16 +24,215 @@ static_assert(mbgl::underlying_type(QMapbox::Feature::PolygonType) == mbgl::unde
...
+*/
 Q_DECL_EXPORT NetworkMode networkMode()
 {
     return static_cast<NetworkMode>(mbgl::NetworkStatus::Get());
 }
 
+/*!
+    \fn void QMapbox::setNetworkMode(QMapbox::NetworkMode mode)
+
+    Forwards the network status \a mode to Mapbox GL Native engine.
```
Comment (tmpsantos) [inline platform/qt/src/qmapbox.cpp]: What are the implications?

File: platform/qt/src/qmapbox.cpp [hunk 12]
```diff
@@ -48,6 +247,12 @@ Q_DECL_EXPORT QList<QPair<QString, QString>>& defaultStyles()
     return styles;
 }
 
+/*!
+    \fn void QMapbox::initializeGLExtensions()
+
+    Initializes the OpenGL extensions required by Mapbox GL Native engine.
+    Must be called once, after an OpenGL context is available.
```
Comment (tmpsantos) [inline platform/qt/src/qmapbox.cpp]: Why?

File: platform/qt/src/qmapbox.cpp [hunk 13]
```diff
@@ -24,16 +24,215 @@ static_assert(mbgl::underlying_type(QMapbox::Feature::PolygonType) == mbgl::unde
...
+    Synonim for QList<quint32> representing a container of annotation identifiers.
+    Reflects mbgl::AnnotationIDs.
+*/
+
+/*!
+    \typedef QMapbox::CustomLayerDeinitializeFunction
+
+    Represents a callback to be called when deinitializing a custom layer.
+
+    Reflects mbgl::style::tializeFunction.
```
Comment (tmpsantos) [inline platform/qt/src/qmapbox.cpp]: Typo.

File: platform/qt/src/qmapbox.cpp [hunk 14]
```diff
@@ -24,16 +24,215 @@ static_assert(mbgl::underlying_type(QMapbox::Feature::PolygonType) == mbgl::unde
...
+
+/*!
+    \typedef QMapbox::AnnotationIDs
+
+    Synonim for QList<quint32> representing a container of annotation identifiers.
+    Reflects mbgl::AnnotationIDs.
+*/
+
+/*!
+    \typedef QMapbox::CustomLayerDeinitializeFunction
```
Comment (tmpsantos) [inline platform/qt/src/qmapbox.cpp]: Add the \warning to CustomLayer stuff. See the existing docs.

File: platform/qt/src/qmapbox.cpp (hunk 1 above)
Comment (brunoabinader) [inline platform/qt/src/qmapbox.cpp]: Will replace that with _alias_ 👍

**Pull request merged**